carma_novatel_driver_wrapper/src/carma_novatel_driver_wrapper.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include <memory>
+
+#include <unordered_set>
+#include "carma_novatel_driver_wrapper/carma_novatel_driver_wrapper.hpp"
+#include <gps_msgs/msg/gps_fix.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+
+using std_msec = std::chrono::milliseconds;
+
+namespace carma_novatel_driver_wrapper
+{   
+    CarmaNovatelDriverWrapper::CarmaNovatelDriverWrapper(const rclcpp::NodeOptions &options, bool auto_init)
+            : CarmaLifecycleNode(options),
+            lifecycle_mgr_(get_node_base_interface(), get_node_graph_interface(), get_node_logging_interface(), get_node_services_interface())
+    {
+        
+        if (auto_init)
+        {
+            initialize();
+        }
+    }
+    
+    void CarmaNovatelDriverWrapper::initialize(){
```

Comment:
Why not just put this in handle_on_configure instead of a separate initialize function which doesn't actually initialize the whole node since parameters are still not loaded.